src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ReferralService.kt

```diff
@@ -312,6 +301,40 @@ class ReferralService(
     return referralRepository.save(referral)
   }
 
+  fun updateDraftReferralDesiredOutcomes(referral: Referral, serviceCategoryId: UUID, desiredOutcomeIds: List<UUID>): Referral {
+    val errors = mutableListOf<FieldError>()
+
+    if (desiredOutcomeIds.isEmpty()) {
+      errors.add(FieldError(field = "desiredOutcomesIds", error = Code.CANNOT_BE_EMPTY))
```

Comment:
small thing: these `FieldError` objects are intended to be used when there are multiple fields being updating in a request. in this case, because we are only ever updating desired outcomes, i think just a plain 400 error is preferable here. it is more consistent with the responses generated for other error conditions below.

Reply:
Ok. I assumed because it's something that is directly related to the user's parameters (rather than indirectly) that we needed to use this type of exception.

I have changed the exception to ServerWebInputException